include/boost/windows_sspi/stream.hpp

```diff
@@ -109,11 +109,11 @@ template <typename NextLayer, typename MutableBufferSequence> struct async_read_
         }
       }
 
-      std::size_t to_return = std::min(net::buffer_size(m_buffer), m_sspi_impl->decrypted_data.size());
-      std::size_t bytes_copied = net::buffer_copy(m_buffer, net::buffer(m_sspi_impl->decrypted_data, to_return));
+      std::size_t to_return = std::min(net::buffer_size(m_buffer), m_sspi_impl->decrypt.decrypted_data.size());
+      std::size_t bytes_copied = net::buffer_copy(m_buffer, net::buffer(m_sspi_impl->decrypt.decrypted_data, to_return));
       boost::ignore_unused(bytes_copied);
       BOOST_ASSERT(bytes_copied == to_return);
-      m_sspi_impl->decrypted_data.erase(m_sspi_impl->decrypted_data.begin(), m_sspi_impl->decrypted_data.begin() + to_return);
+      m_sspi_impl->decrypt.decrypted_data.erase(m_sspi_impl->decrypt.decrypted_data.begin(), m_sspi_impl->decrypt.decrypted_data.begin() + to_return);
```

Comment:
It makes me a little sad to see a vector erase operation in a loop - it's probably immaterial in practice, but theoretically I wonder if there's a cleaner way (circular buffer, segmented buffer, decrypt directly into client buffer?)